core/src/main/scala/cats/Traverse.scala

```diff
@@ -213,6 +213,22 @@ trait Traverse[F[_]] extends Functor[F] with Foldable[F] with UnorderedTraverse[
 
   override def unorderedSequence[G[_]: CommutativeApplicative, A](fga: F[G[A]]): G[F[A]] =
     sequence(fga)
+
+  def foldTraverse[A, B, G[_]](fa: F[A])(f: A => G[B])(implicit G: Monad[G], M: Monoid[B]): G[B] =
+    foldM(fa, M.empty) { case (acc, a) =>
+      G.map(f(a)) { b =>
+        M.combine(acc, b)
+      }
+    }
```

Comment:
I don't like needing `Monad` here, since `Applicative` should be enough using `foldLeft` and `map2`
But, it seems the implementation of `foldM` is reasonably optimized and ensures an early stop, not sure if using `map2` would achieve that as well.

What do you all think?
Any feedback is welcome.

Reply:
I think you may be able to use `map2Eval`? See also:
- https://github.com/typelevel/cats/issues/4408